kolibri/utils/version.py

```diff
@@ -305,20 +335,16 @@ def get_prerelease_version(version):
 
     # Calculate suffix...
 
-    # If a description from git is available and we haven't already
-    # found a suffix
-    if not suffix and tag_describe:
```

Comment:
Not sure of the logic when replacing `if not suffix` - essentially, the suffix is only written when in `alpha-0`. I can't read the fallback behavior in the code anymore? It's supposed to be that whenever we're in `alpha-0` stage, the suffix is automatically generated as some `.dev-XYZ` suffix with date and git commit hash present.

Reply:
The issue here is that a suffix is always generated (because it just generates a suffix based on the timestamp of the most recent git commit), because of the practice of not encoding pre-releases in our version file.

Reply:
Will add some extra test cases to make some guarantees about behaviour here, and what we expect.

Reply:
It only generates the suffix (`.dev`) if it's alpha 0, which should be expected since that's anything up until the first pre-release (alpha 1), see: `if version[4] == 0 and version[3] == 'alpha':`

Reply:
See also docstring: https://github.com/learningequality/kolibri/pull/2784/files#diff-766a373b5576b8fcac0456888d19294cR315

Reply:
The issue has been that we have not, historically, increased the version beyond alpha-0 except to do a final release. So just checking that there is no suffix has not allowed us to generate a git tag based release that encodes its version properly.